src/vs/workbench/contrib/terminalContrib/suggest/common/terminalSuggestConfiguration.ts

```diff
@@ -118,16 +120,27 @@ export const terminalSuggestConfiguration: IStringDictionary<IConfigurationPrope
 	},
 	[TerminalSuggestSettingId.RunOnEnter]: {
 		restricted: true,
-		markdownDescription: localize('suggest.runOnEnter', "Controls whether suggestions should run immediately when `Enter` (not `Tab`) is used to accept the result."),
-		enum: ['ignore', 'never', 'exactMatch', 'exactMatchIgnoreExtension', 'always'],
+		markdownDescription: localize('suggest.runOnEnter', "Controls whether suggestions should run immediately when `Enter` (not `Tab`) is used to accept the result. Also see {0}.", `\`#${TerminalSuggestSettingId.SelectionMode}#\``),
+		enum: ['never', 'exactMatch', 'exactMatchIgnoreExtension', 'always'],
 		markdownEnumDescriptions: [
-			localize('runOnEnter.ignore', "Ignore suggestions and send the enter directly to the shell without completing. This is used as the default value so the suggest widget is as unobtrusive as possible."),
 			localize('runOnEnter.never', "Never run on `Enter`."),
 			localize('runOnEnter.exactMatch', "Run on `Enter` when the suggestion is typed in its entirety."),
 			localize('runOnEnter.exactMatchIgnoreExtension', "Run on `Enter` when the suggestion is typed in its entirety or when a file is typed without its extension included."),
 			localize('runOnEnter.always', "Always run on `Enter`.")
 		],
-		default: 'ignore',
+		default: 'never',
+		tags: ['preview']
+	},
+	[TerminalSuggestSettingId.SelectionMode]: {
+		markdownDescription: localize('terminal.integrated.selectionMode', "Controls how suggestion selection works in the integrated terminal. Also see {0}.", `\`#${TerminalSuggestSettingId.RunOnEnter}#\``),
+		type: 'string',
+		enum: ['partial', 'always', 'never'],
+		enumDescriptions: [
+			localize('terminal.integrated.selectionMode.partial', "Will show a border until navigation has occurred. After that, it will show selection."),
```

Comment:
We should talk about how tab and enter behave as that's the main difference. Reading this it looks like it only impacts looks.